### PR TITLE
0.5 image

### PR DESCRIPTION
**Pull request by jeremybosma**
This pr introduces additional functionality for local src/ folder images for the astro component in our own image component

### CODE REVIEW

File: src/components/Image.astro [hunk 1]
```diff
@@ -42,29 +50,47 @@ const {
...
-          as="img"
-          {frame}
-          {position}
-          {radius}
-          {...rest}
-        />
-      )}
-      {width && height && (
+      {isLocalAsset ? (
         <AstroImage
```
Comment (silveltman) [inline src/components/Image.astro:54]: You use the AstroImage twice, here and below. You can just include in in the check, something like
`isLocalAsset || (!isLocalAsset && width && height)`

File: src/components/Image.astro [hunk 2]
```diff
@@ -34,6 +34,14 @@ const {
   ...rest
 } = Astro.props
 
+const isLocalAsset = src && (src.startsWith('*/') || src.startsWith('../'))
```
Comment (silveltman) [inline src/components/Image.astro:37]: There is a better and more reliable way to check if something is a URL (so not-local). Know know from the top of my head, but look it up

File: src/components/Image.astro [hunk 3]
```diff
@@ -34,6 +34,14 @@ const {
...
+let localImage
+
+if (isLocalAsset) {
+  const { default: importedImage } = await import(/* @vite-ignore */ src)
```
Comment (silveltman) [inline src/components/Image.astro:41]: What does the @vite-ignore do? Is this a hacky workaround for a problem that can be fixed?

File: src/components/Image.astro [hunk 4]
```diff
@@ -42,29 +50,47 @@ const {
...
         />
+      ) : (
+        <>
+          {!width && !height && (
+            <Root
+              class:list="image"
+              if={src}
+              {src}
+              {alt}
+              as="img"
```
Comment (silveltman) [inline src/components/Image.astro:75]: re-order this one to the top

**Review by silveltman — CHANGES_REQUESTED** [review]

File: src/components/Image.astro (hunk 2 above)
Comment (jeremybosma) [reply src/components/Image.astro]:
```suggestion
const isUrl = (str: string): boolean => {
```

File: src/components/Image.astro (hunk 3 above)
Comment (jeremybosma) [reply src/components/Image.astro]: const importedImage = await import(`../assets/${src}`)

**Review by silveltman — APPROVED** [review]

**Pull request merged**